test/alternator/conftest.py

```diff
@@ -59,6 +59,46 @@ def pytest_collection_modifyitems(config, items):
         if "veryslow" in item.keywords:
             item.add_marker(skip_veryslow)
 
+# When testing Alternator running with --alternator-enforce-authorization=1,
+# we need to find a valid username and secret key to use in the connection.
+# Alternator allows any CQL role as the username any CQL role, and the key
+# is that role's password's salted hash. We can read a valid role/hash
+# from the appropriate system table, but can't do it with Alternator (because
+# we don't know yet the secret key!), so we need to do it with CQL.
+get_valid_alternator_role_cache = {}
+def get_valid_alternator_role(url):
+    if url in get_valid_alternator_role_cache:
+        return get_valid_alternator_role_cache[url]
+    import cassandra.cluster
```

Comment:
Any particular reason the imports are inline ? And not on the top ? to speed up test collection ?

Reply:
Both this unorthodox Python style choice, and fact I didn't split this into functions (as you suggested above) stems from the same reason: I really wanted the CQL-dependence of this code to be segragated into one function `get_valid_alternator_role()`, and not starting to split it to many different places in the code. The fact that the Alternator test depends on CQL, and on the Cassandra libraries, is unfortunate, and somewhat unexpected, so I really wanted to segregate it into just one place.

Reply:
as for conftest, it might be a good practice, if the imports are slow or have side-effects (python-driver ones shouldn't, and we should fix if they do)

but keeping everything together is a valid enough case,
only small downside it might take more time to fail, on import failure, but it's probably not that important

Reply:
By the way, that "import failure" is actually another benefit of my approach: I think (?) it means that if we don't reach this code, it doesn't matter that the library is missing. So for example someone can run alternator pytest against Amazon DynamoDB ("pytest --aws") without needing to install the Cassandra library which is completely irrelevant for DyanamoDB.